src/cpp/include/openvino/genai/streamer_base.hpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "openvino/genai/tokenizer.hpp"
+
+namespace ov {
+
+/** 
+ * @brief base class for streamers. In order to use inherit from from this class and inplement put, and methods
+ * 
+ * @param m_tokenizer tokenizer
+*/
+class StreamerBase {
+public:
+    Tokenizer m_tokenizer;
```

Comment:
That would be annoying to access the protected field. At the same time, a user already have access to it because they could have provided it in constructor. Public just lets them access it in easier way